### DISCUSSION TITLE
Dynamic Groups

### DISCUSSION THREAD

**Discussion by kdrwygvh** [almenscorner/IntuneCD#196] [Ideas]
Is it possible/feasible to include dynamic user and device groups in a backup?

**Comment by almenscorner** [comment]
Dynamic groups are backed up as part of a configuration or do you mean groups standalone that are not assigned to anything in Intune?

**Comment by almenscorner** [comment]
For example:

```json
     "assignments": [
          {
               "source": "direct",
               "target": {
                    "@odata.type": "#microsoft.graph.groupAssignmentTarget",
                    "deviceAndAppManagementAssignmentFilterId": null,
                    "deviceAndAppManagementAssignmentFilterType": "none",
                    "groupName": "test",
                    "groupType": "DynamicMembership",
                    "membershipRule": "(device.enrollmentProfileName -eq \"shareddevice\")"
               }
          }
     ]
```

**Reply by kdrwygvh** [reply]
Ah I see. So in your example would the test group be created as new in a target tenant?

**Reply by almenscorner** [reply]
If you include the `--create-groups` argument, yes, it should be created

**Comment by kdrwygvh** [comment]
Very good, thank you for clarifying. Will close the discussion as resolved.
